### PR TITLE
Use streams-like interfaces instead of methods

### PR DESCRIPTION
**Pull request by ghost**
I don't know if this will be merged, but I just wanted to not use methods (that's also better for ARC which works with Nimes - it didn't work but I submitted a bug report which got it fixed :P https://github.com/nim-lang/Nim/issues/14159).

This also makes nimes work with mostly same performance in ARC as with the default GC (by default ARC as of now uses string comparisons for methods to work across DLLs and stuff which is really slow for nimes since it has to decide which method to call on each iteration).

Even if you won't merge it's fine, and if you do - please squash merge since I messed up history a bit :)

### CODE REVIEW

File: src/nes.nim [hunk 1]
```diff
@@ -36,4 +36,4 @@ proc run*(nes: NES, seconds: float) =
     cycles -= nes.step()
 
 proc buffer*(nes: NES): var Picture =
-  nes.ppu.front
+  nes.ppu.front
```
Comment (def-) [inline src/nes.nim]: please add a newline here and in the other files, rest of the change looks fine.

**Review by def- — CHANGES_REQUESTED** [review]
Generally I'm fine, thanks!

File: src/nes.nim (hunk 1 above)
Comment (ghost) [reply src/nes.nim]: @def- done

File: src/nes/cartridge.nim [hunk 2]
```diff
@@ -4,7 +4,7 @@ export types.Cartridge
 
 const iNESMagic = 0x1A53454E
 
-type iNESHeader = object {.packed.}
+type iNESHeader {.packed.} = object
```
Comment (ghost) [inline src/nes/cartridge.nim:7]: ``/home/dian/Projects/nimes/src/nes/cartridge.nim(7, 26) Warning: type pragmas follow the type name; this form of writing pragmas is deprecated [Deprecated]``

**Comment by def-** [comment]
Thanks!

**Pull request merged**